test/unit/sql/where.test.ts

```diff
@@ -2246,6 +2246,39 @@ describe(getTestDialectTeaser('SQL'), () => {
 
     if (sequelize.dialect.supports.JSONB) {
       describe('JSONB', () => {
+        if (dialectSupportsArray()) {
+          testSql({
+            jsonbAttr: {
+              [Op.anyKeyExists]: ['a', 'b'],
+            },
+          }, {
+            default: `[jsonbAttr] ?| ARRAY['a', 'b']`,
+          });
+
+          testSql({
+            jsonbAttr: {
+              [Op.allKeysExist]: ['a', 'b'],
+            },
+          }, {
+            default: `[jsonbAttr] ?& ARRAY['a', 'b']`,
+          });
+
+          testSql({
+            jsonbAttr: {
+              [Op.anyKeyExists]: literal(`SELECT jsonb_array_elements_text('ARRAY["a","b"]')`),
+            },
+          }, {
+            default: `[jsonbAttr] ?| ARRAY(SELECT jsonb_array_elements_text('ARRAY["a","b"]'))`,
+          });
```

Comment:
When using `literal`, we don't assume what's in the literal. It's the responsibility of the user to use a valid literal.

```suggestion
          testSql({
            jsonbAttr: {
              [Op.anyKeyExists]: literal(`ARRAY(SELECT jsonb_array_elements_text('ARRAY["a","b"]'))`),
            },
          }, {
            default: `[jsonbAttr] ?| ARRAY(SELECT jsonb_array_elements_text('ARRAY["a","b"]'))`,
          });
```

Reply:
Since we will wrap the literal within an array, I think we will skip this commit suggestion https://github.com/sequelize/sequelize/pull/15073#discussion_r990766377

Reply:
You'll need to test both `[literal()]` and `literal`, both are valid

Reply:
I see